sync/class.jetpack-sync-functions.php

```diff
@@ -138,6 +168,31 @@ public static function get_protocol_normalized_url( $callable, $new_value ) {
 		return set_url_scheme( $new_value, $forced_scheme );
 	}
 
+	public static function get_raw_url( $option_name ) {
+		global $wpdb;
+
+		$value = null;
+		if ( 'home' == $option_name && Jetpack_Constants::is_defined( 'WP_HOME' ) ) {
+			$value = Jetpack_Constants::get_constant( 'WP_HOME' );
+		} else if ( 'siteurl' == $option_name && Jetpack_Constants::is_defined( 'WP_SITEURL' ) ) {
+			$value = Jetpack_Constants::get_constant( 'WP_SITEURL' );
+		} else {
+			// Let's get the option from the database so that we can bypass filters. This will help
+			// ensure that we get more uniform values.
+			$value = Jetpack_Sync_Options::get_option( $option_name );
+		}
+
+		if ( is_ssl() ) {
```

Comment:
I don't think it would cause issues since we have the ssl normalization logic now. But, if we stick with this raw approach, then I think we can get rid of the ssl normalization. So, I think I agree with you that we could probably get rid of this. 😉